index.d.ts

```diff
@@ -42,6 +48,11 @@ export type AnyEnhancer<State = unknown> =
   ActionEnhancer<Enhancement, State, any> |
   ActionEnhancer<Enhancement, State, never>;
 
+declare function isEnhancerById<E extends Enhancement, State = unknown, Value = never>(e: ActionEnhancer<E, State, Value>): e is ActionEnhancerById;
```

Comment:
Probably want to pass the generics into the return. `e is ActionEnhancerById<E, State, Value>`